### PR TITLE
Updates for week of May 15

### PR DESCRIPTION
**Pull request by duglin**
- clarify how to use `typemap`
  - only use the "media-type" (no parameters) of the `contenttype` value
  - set `contenttype` automatically when Resource doc is inlined (sometimes)
  - if more than one key in `typemap` matches, and they're not the same value, the use `binary`
  - define a `binary` type to force using `RESOURCEbase64` if needed

Fixes: #116

### CODE REVIEW

File: core/spec.md [hunk 1]
```diff
@@ -1906,30 +1906,56 @@ The following describes the attributes of Registry model:
...
         can be serialized as a "string" with some escaping.<br>
+
     For some well-known `contenttype` values (e.g. `application/json`) the
     first case can be easily determined by the server. However, for custom
     values the server will need to be explicitly told to when to consider it
     to be of the correct type. The `typemap` attribute allows for this by
     defining a mapping of `contenttype` values to well-known xRegistry format
     types.
 
+    Since the `contenttype` value is a "media-type" per
```
Comment (jskeet) [inline core/spec.md:1917]: Remembering adding this to the CloudEvents spec recently, maybe emphasize that this has to be done case-insensitively? I see this is mentioned later on, but it feels useful to have everything to do with lookup specified in one place.

**Review by jskeet — COMMENTED** [review]
A couple of nits, but otherwise seems fine from my perspective of "not really following xRegistry much" :)

File: core/spec.md (hunk 1 above)
Comment (duglin) [reply core/spec.md:1917]: done

File: core/spec.md [hunk 2]
```diff
@@ -3465,6 +3488,19 @@ Where:
...
     3 attributes (and any associated data).
+  - When `RESOURCE` is present, the server MAY choose to modify non-semantic
+    significant characters. For example, to remove (or add) whitespace. In
+    other words, there is no requirement for the server to persist the
+    document in the exact byte-for-byte format in which it was provided. If
+    that is desired then `RESOURCEbase64` MUST be used instead.
+  - On a `PUT` or `POST`, when `RESOURCE` is present, if no `contenttype`
+    value is provided then the server MUST set it to same type as the incoming
+    request, e.g. `application/json`, even if the entity previous had a
+    contenttype` value.
```
[the remark this answers is not in the document]
Comment (duglin) [reply core/spec.md:3499]: done

**Comment by duglin** [comment]
Approved on the 5/16 call

**Pull request merged**